install/kubernetes/karpenter-provisioner.yaml.tpl

```diff
@@ -0,0 +1,79 @@
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  subnetSelector:
+    karpenter.sh/discovery: ${CLUSTER_NAME}
+  securityGroupSelector:
+    karpenter.sh/discovery: ${CLUSTER_NAME}
+---
+# https://karpenter.sh/docs/getting-started/getting-started-with-karpenter/
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: gpu
+spec:
+  provider:
+    instanceProfile: eksctl-KarpenterNodeInstanceProfile-${CLUSTER_NAME}
+    subnetSelector:
+      karpenter.sh/discovery: ${CLUSTER_NAME}
+    securityGroupSelector:
+      karpenter.sh/discovery: ${CLUSTER_NAME}
+  ttlSecondsAfterEmpty: 30
+  consolidation:
+    enabled: true
+  taints:
+    - key: nvidia.com/gpu
+      value: "true"
+      effect: NoSchedule
+  requirements:
+    - key: karpenter.sh/capacity-type
+      operator: In
+      values: ["spot"]
+    - key: node.kubernetes.io/instance-type
+      operator: In
+      values:
+        # aws ec2 describe-instance-types --region us-west-2 --query "InstanceTypes[?GpuInfo!=null].InstanceType" --output json | jq -r '.[]' | sort | grep -v dl1 | grep -v inf | grep -v p5 | grep -v trn1 | awk '{print "\""$1"\","}'
```

Comment:
If you want you can use this instead of having to list them all out
```
- key: karpenter.k8s.aws/instance-category
          operator: In
          values: ["g", "p"]
- key: karpenter.k8s.aws/instance-family
          operator: NotIn
          values: ["p5"]
```

Reply:
This is a good simplification and one that we'd need to extend a little further to exclude graviton architecture hosts (our images aren't produced or tested against arm yet). I'll cook that up. Thanks!